app/main.py

```diff
@@ -1,25 +1,74 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
```

Comment:
Do not bind `owner` here and `instance`/`owner` from `__get__` to only the `SlideLimitationValidator` class, the `IntegerRange` class should be reusable across different classes. Replace `SlideLimitationValidator` with `Any`.